### PR TITLE
Fix build of compiler-rt on FreeBSD

### PR DESCRIPTION
**Pull request by asomers**
Broken since ee6011fc71e02485f2dffcc25be64631c2008775 removed cmake from the
process.  There are likely other platforms still broken, but I didn't test on them.

### CODE REVIEW

**Comment by TimNN** [comment]
This fix should probably be applied to rustbuild as well, the relevant function is [this one](https://github.com/rust-lang/rust/blob/master/src/bootstrap/native.rs#L158) as far as I can tell.

File: mk/rt.mk [hunk 1]
```diff
@@ -384,7 +384,11 @@ COMPRT_OBJS_$(1) += emutls.o
 endif
 
 ifeq ($$(findstring msvc,$(1)),)
+
+ifeq ($$(findstring freebsd,$(1)),)
```
Comment (nagisa) [inline mk/rt.mk]: err, this condition is inside one which checks for `msvc`. Pretty sure there’s no combination of freebsd and msvc out there.

Comment (TimNN) [inline mk/rt.mk]: I believe these conditions are Makefile-speak vor "not contains" (`findstring` returns the found string, which is then compared to the empty string, as far as I know).

Comment (asomers) [inline mk/rt.mk]: Correct, the first condition checks for "not msvc" and the second checks for "not freebsd".

**Comment by alexcrichton** [comment]
@bors: r+ 8604c5494eed8d111757f23a874d9ce8ca673e2c

Thanks @asomers!

**Comment by Manishearth** [comment]
Travis fails

**Comment by TimNN** [comment]
@Manishearth that failure is unrelated to this pr and has recently been fixed (#35014).

**Comment by Manishearth** [comment]
Yeah, just noticed 😄

**Pull request merged**